Templating/TemplateGuesser.php

```diff
@@ -55,10 +64,21 @@ public function guessTemplateName($controller, Request $request, $engine = 'twig
     {
         $className = class_exists('Doctrine\Common\Util\ClassUtils') ? ClassUtils::getClass($controller[0]) : get_class($controller[0]);
 
-        if (!preg_match('/Controller\\\(.+)Controller$/', $className, $matchController)) {
-            throw new \InvalidArgumentException(sprintf('The "%s" class does not look like a controller class (it must be in a "Controller" sub-namespace and the class name must end with "Controller")', get_class($controller[0])));
+        $matchController = null;
+        foreach ($this->controllerPatterns as $pattern) {
+            if (preg_match($pattern, $className, $tempMatch)) {
+                $matchController = $tempMatch;
+                break;
+            }
+        }
+        if (null === $matchController) {
+            throw new \InvalidArgumentException(sprintf('The "%s" class does not look like a controller class (its FQN must match one of the following regexps: "%s")', get_class($controller[0]), implode('", "', $this->controllerPatterns)));
         }
-        if (!preg_match('/^(.+)Action$/', $controller[1], $matchAction)) {
+
+        if ($controller[1] === '__invoke') {
+            $matchAction = $matchController;
+            $matchController = null;
+        } elseif (!preg_match('/^(.+)Action$/', $controller[1], $matchAction)) {
```

Comment:
With DunglasActionBundle, the preferred method name is `__invoke` and, as it's recommended to have only 1 action per class, the `Action` suffix is not mandatory in methods name. This test should be removed.